File: c++/KillProcess.cpp
class Solution {
public:
    vector<int> killProcess(vector<int>& pid, vector<int>& ppid, int kill) {
        unordered_map<int, vector<int>> adjList;
        vector<int> output;
        
        for (int index = 0; index < pid.size(); ++index) {
            int parent = ppid[index];
            adjList[parent].push_back(pid[index]);
        }
        
        queue<int> q;
        q.push(kill);
        
        while (q.size()) {
            int curLen = q.size();
            for (int i = 0; i < curLen; ++i) {
                int curPid = q.front();
                output.push_back(curPid);
                q.pop();
                
                for (auto child : adjList[curPid]) {
                    q.push(child);
                }
            }
        }
        
        return output;
    }
};

File: python3/KillProcess.py
class Solution:
    def killProcess(self, pid: List[int], ppid: List[int], kill: int) -> List[int]:
        adjList = {}
        
        for index, child in enumerate(pid):
            parent = ppid[index]
            if parent not in adjList:
                adjList[parent] = [child]
            else:
                adjList[parent].append(child)
        
        output = []
        queue = [kill]
        
        while (len(queue)):
            curLen = len(queue)
            for i in range(curLen):
                curPid = queue.pop(0)
                output.append(curPid)
                
                if curPid in adjList:
                    for child in adjList[curPid]:
                        queue.append(child)
        
        return output
